frontend/lib/resolution/copy-elision.cpp

```diff
@@ -599,6 +613,11 @@ void FindElidedCopies::propagateChildToParent(VarFrame* frame, VarFrame* parent,
         if (state.lastIsCopy && frame->eligibleVars.count(id) == 0) {
           CopyElisionState& parentState = parent->copyElisionState[id];
           parentState.lastIsCopy = true;
+
+          // If there are valid copies in a child frame, the parent's
+          // accumulated points thus far are no longer valid.
+          parentState.points.clear();
```

Comment:
I think this is probably right, but I'm not sure this is happening in the right place. Consider the case of two branches of a conditional both calling `propagateChildToParent`. In that case, we'd like `parentState.points` to include the points within the `if` and `else` clauses. I'm talking about something like this:

``` chapel
proc bar() {
  var x = makeRecord();
  if something {
     var y1 = x;
  } else {
     var y2 = x;
  }
  // now x should be considered 'dead' as it was copy elided from
  // in both branches
}
```

I would imagine that the CopyElisionState for the body of `bar` should have the IDs of `var y1 = x` and `var y2 = x` both.

Reply:
I believe the way this works in practice is that we are actually propagating from a single frame representing the entire conditional, rather than each branch (which was addressed by ``handleDisjunction``). So I don't think we're in danger of successive calls to this method erasing things.

I'll spend some time trying to see if I can create a case that breaks this though. Maybe with some else-ifs?

Reply:
Ah, yes, propagateChildToParent seems to be only called in non-conditional cases / after the special logic for the conditionals. I think it's fine. Thanks for thinking about it.

Reply:
(A comment on that method wouldn't hurt though)